### DISCUSSION TITLE
How can I publish my packages to npm?

### DISCUSSION THREAD

**Discussion by melishev** [vercel/turborepo#910] [Help] [answered]
Hello, I have a project written in turborepo, it has 5 publishing packages and 2 websites.

I used to use semantic-release to publish my repositories, but it's not ready to work with monorepo.

On the page [Turborepo Tools](https://turborepo.org/docs/guides/monorepo-tools) , I saw the [auto](https://github.com/intuit/auto) library, for convenient publication of a monorepo. But she doesn't see anything but the main package.json

Please tell me options for publishing my mono-repository, I really liked [auto](https://github.com/intuit/auto) , but unfortunately it does not work properly.

**Comment by MontoyaAndres** [comment]
any news?

**Comment by biswarup35** [comment]
Hey @melishev 
I'm also using Turborepo and semantic-release for publishing my NPM packages. The problem I'm currently facing is that semantic release has a one-to-one relationship with the repository. That's the reason I'm unable to publish multiple packages using these two tools. Even if we use semantic-release for each package it will override/collide with the git-tag. I am able to publish one package using these two tools. If you found any solution please do share it with me, thanks.

**Comment by nathanhammond** [comment, marked as answer]
We recommend using https://github.com/changesets/changesets for publishing packages and Vercel for deploying websites.

**Reply by biswarup35** [reply]
Thanks man

**Reply by gaspar09** [reply]
In case this is useful, we have an example of this setup as well https://github.com/vercel/turborepo/tree/main/examples/with-changesets

**Reply by Sylvenas** [reply]
Hi,@gaspar09, how to deal with dependencies between packages, for example, `@acme/core` depends on `@acme/utils`, and after publishing to npm, `@acme/utils` will not be found, because the dependency address of `@acme/utils` is `workspace:* ` instead of `@acme/utils`.

**Reply by gaspar09** [reply]
@Sylvenas , you can publish `@acme/utils` as well.

The nice thing about changesets is that when you make a changeset for `@acme/utils`, the workflow will know to that the change affects `@acme/core` and will re-publish both with updated versions.

Alternatively, you can bundle `@acme/utils` into the `@acme/core` but that involves updating the build command for `@acme/core`.

**Reply by Sylvenas** [reply]
After trying, `changesets` will automatically replace `workspace:*` with the version number, great!

**Comment by iamnewton** [comment]
Is this tested?  You are only calling the publish task and not the `version-packages` one.  Do you have it auto-publish to NPM or does this open a PR?
